Update extension to work with Windows

dclemmon opened the pull request:
This pull request adds the ability to use this inkscape extension under windows.  I've tested it with Windows
10, and it works as described in the readme file.  I have not been able to test it under Linux.  I encountered issues using the `os.path.expanduser('~<user>')` and `os.path.expanduser('~')` syntax, where it worked well in the python shell, but under inkscape, gave the incorrect path.

Commit:
Updates extension to work with Windows

Commit:
Matches code style

olibia commented:
Hi, thank you for the PR.

I tried the following and it worked on Windows 10.
```python
if sys.platform == 'win32':
    path = os.path.expanduser('~/AppData/Roaming/inkscape/palettes')
else:
    path = os.path.expanduser('~/.config/inkscape/palettes')
```
Is getpass necessary for some reason?

dclemmon commented:
In my general testing the `os.path.expanduser('~/AppData/Roaming/inkscape/palettes')` syntax worked, but once I installed the plugin in Inkscape, it was unable to determine the correct path.  I found that `getpass` returned the correct user while running under Inkscape, and that I could create the correct path from that data.

dclemmon commented:
After a little thinking, I'm testing on a corporately controlled laptop.  The issue in testing might be related to policies changing the values of `%HOMEDRIVE%` and `%HOMEPATH%` to something that changes the behavior of `os.path.expanduser()`.  Let me know if you are unable to reproduce the issue.

olibia commented:
I tried both ways and it works. I just didn't know what difference getpass made. I'm hesitating on using the `'C:/Users/%s/AppData/Roaming/inkscape/palettes'` path as it is hard coded.

Can you try the following? I tested it on Windows too and it worked.
```python
def palettes_path(self):
    if sys.platform == 'win32':
      path = os.path.join(os.environ['APPDATA'], 'inkscape', 'palettes')
    else:
      path = os.environ.get('XDG_CONFIG_HOME', '~/.config')
      path = os.path.join(path, 'inkscape', 'palettes')

    return os.path.expanduser(path)
```

Commit:
Updates palettes path method with more robust path creation, removes getpass and sys dep

dclemmon commented:
I agree, the hard path is a bad idea, your suggestion is testing well.  I've updated the PR with your suggestions, and I've removed `sys` and `getpass` as well

olibia commented:
Great, thanks again!
